sqlglot/parser.py

```diff
@@ -4993,7 +4993,14 @@ def _parse_colon_as_variant_extract(
                 end_token = self._prev
 
             if path:
-                json_path.append(self._find_sql(self._tokens[start_index], end_token))
+                path_sql = self._find_sql(self._tokens[start_index], end_token)
+
+                # Escape single quotes from Snowflake's colon extraction (e.g. col:"a'b") as
+                # it'll roundtrip to a string literal in GET_PATH
+                if path_sql[0] == '"':
+                    path_sql = path_sql.replace("'", "\\'")
```

Comment:
I think we can do `self.dialect.generator.escape_str(path_sql)`, which is preferable vs hard-coding the escape.

Reply:
That was my intuition as well but `generator` is not a property so we'd have to construct a new Generator for each parsed path:

`path_sql = self.dialect.generator().escape_str(path_sql)`

Should I go ahead and change it anyways?

Reply:
Ah I see, no then we don't want to instantiate a new generator every time. Can we just make this more generic instead of hardcoding the `\\'`?

Reply:
How about the latest approach?

Reply:
LGTM